### PR TITLE
Feature/async query polling

### PR DESCRIPTION
**Pull request by niross**
- On query submit create the show a "query submitting" page and start polling for results
    - Polling starts at 0.5 seconds and ramps up to a max of 5
- Updates `execute_query_sync` to create a query log before running the query asynchronously
- Reverts async download as it's covered by a separate ticket.
- Skip async download tests as that feature is still in development (in a separate ticket)
- Configurable via the `RUN_ASYNC_QUERIES` waffle flag

You can use pg_sleep to test locally: `select 123 as a_col from (select pg_sleep(10)) a;`

### CODE REVIEW

File: dataworkspace/dataworkspace/static/explorer_async.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+function pollForQueryResults(queryLogId, delay, delayStep, maxDelay) {
+  var resultsContainer = document.getElementById('query-results');
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', '/data-explorer/logs/' + queryLogId + '/results-json/');
+  xhr.onreadystatechange = function() {
+    if (this.readyState === XMLHttpRequest.DONE) {
+      var resp = JSON.parse(xhr.responseText);
+      resultsContainer.innerHTML = resp.html;
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/static/explorer_async.js:8]: As per slack: let's only do this for state 1+2

File: dataworkspace/dataworkspace/static/explorer_async.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      if (resp.state === 0) {
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/static/explorer_async.js:9]: As per slack, let's use JS variables here to give these semantic meaning, e.g. QUERY_STATE_RUNNING/FAILED/COMPLETE.

**Review by samuelhwilliams — APPROVED** [review]

**Review by abbas123456 — APPROVED** [review]
Looks good

File: dataworkspace/dataworkspace/apps/explorer/templates/explorer/partials/query_submitting.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="govuk-grid-row" id="async-query-submitting">
+  <div class="govuk-!-margin-bottom-7 loading-spinner"></div>
+  <p class="govuk-heading-m" style="text-align: center;">
+    Your query is being submitted to Data Explorer for execution.
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/apps/explorer/templates/explorer/partials/query_submitting.html:4]: Ah - per the prototype, this content should be "Your query is currently being executed by Data Explorer."

The "submitted" line is from the ajax "run" button. While the page is doing a full reload, we know that the query has already been submitted.

Comment (niross) [reply dataworkspace/dataworkspace/apps/explorer/templates/explorer/partials/query_submitting.html:4]: oh right ok I will update

**Review by samuelhwilliams — APPROVED** [review]
Nice 👍

**Pull request merged**